gridstatus/ercot.py

```diff
@@ -3386,10 +3392,13 @@ def read_doc(
     ):
         logger.debug(f"Reading {doc.url}")
 
-        response = requests.get(doc.url, **(request_kwargs or {})).content
-        df = pd.read_csv(
-            io.BytesIO(response), compression="zip", **(read_csv_kwargs or {})
-        )
+        if request_kwargs:
+            response = requests.get(doc.url, **(request_kwargs or {})).content
+            df = pd.read_csv(
+                io.BytesIO(response), compression="zip", **(read_csv_kwargs or {})
+            )
+        else:
+            df = pd.read_csv(doc.url, compression="zip", **(read_csv_kwargs or {}))
```

Comment:
This change is necessary because `doc.url` might be a `StringIO`. `pd.read_csv` can read in from a file buffer, so, unless we need to use `requests` with `request_kwargs`, we can use `pd.read_csv` to handle both urls and `StringIO`.